test/Recollateralization.test.ts

```diff
@@ -3088,6 +3113,141 @@ describe(`Recollateralization - P${IMPLEMENTATION}`, () => {
+          await backingManager.rebalance(TradeKind.DUTCH_AUCTION)
+          const trade = await ethers.getContractAt(
+            'DutchTrade',
+            await backingManager.trades(token0.address)
+          )
+          await token1.connect(addr1).approve(trade.address, initialBal)
+
+          const start = await trade.startTime()
+          const end = await trade.endTime()
+
+          // Simulate 5 minutes of blocks, should swap at right price each time
+          for (let now = await getLatestBlockTimestamp(); now < end; now += 12) {
+            const actual = await trade.connect(addr1).bidAmount(now)
+            const expected = divCeil(
+              await dutchBuyAmount(
+                fp(now - start).div(end - start),
+                rTokenAsset.address,
+                collateral0.address,
+                issueAmount,
+                config.minTradeVolume,
+                config.maxTradeSlippage
+              ),
+              bn('1e12') // fix for decimals
+            )
+            expect(actual).to.equal(expected)
+
+            const staticResult = await trade.connect(addr1).callStatic.bid()
+            expect(staticResult).to.equal(expected)
+            await advanceToTimestamp((await getLatestBlockTimestamp()) + 12)
+          }
+        })
+
+        it('Should handle no bid case correctly', async () => {
+          await backingManager.rebalance(TradeKind.DUTCH_AUCTION)
+          const trade = await ethers.getContractAt(
+            'DutchTrade',
+            await backingManager.trades(token0.address)
+          )
+          await token1.connect(addr1).approve(trade.address, initialBal)
+          await advanceToTimestamp((await getLatestBlockTimestamp()) + auctionLength - 1)
+          await expect(
+            trade.connect(addr1).bidAmount(await getLatestBlockTimestamp())
+          ).to.be.revertedWith('auction over')
+          await expect(trade.connect(addr1).bid()).be.revertedWith('auction over')
+
+          // Should be able to settle
+          await expect(trade.settle()).to.be.revertedWith('only origin can settle')
+          await expect(backingManager.settleTrade(token0.address))
+            .to.emit(backingManager, 'TradeSettled')
+            .withArgs(trade.address, token0.address, token1.address, 0, 0)
+
+          // Should NOT start another auction, since caller was not DutchTrade
+          expect(await backingManager.tradesOpen()).to.equal(0)
+        })
+
+        it('Should bid in final second of auction and launch another auction', async () => {
+          await backingManager.rebalance(TradeKind.DUTCH_AUCTION)
+          const trade = await ethers.getContractAt(
+            'DutchTrade',
+            await backingManager.trades(token0.address)
+          )
+          await token1.connect(addr1).approve(trade.address, initialBal)
+
+          // Snipe auction at 1s left
+          await advanceToTimestamp((await getLatestBlockTimestamp()) + auctionLength - 3)
```

Comment:
curious, why not - 2?

Reply:
`rebalance()` occurs at 0/300; `token1.approve()` occurs at 1/300; it fasts forward 297s to get to 298/300; and then `bid()` executes at 299/300